src/components/BookShelf/AddBookModal/AddBookForm/AddBookForm.tsx

```diff
@@ -0,0 +1,194 @@
+import {
+  Icon,
+  IconEnum,
+  ImageField,
+  TextAreaField,
+  TextField,
+  UIbutton,
+} from "@/src/components";
+import { Form, Formik } from "formik";
+import { FC, useState } from "react";
+import styles from "../AddBookModal.module.scss";
+import { validationSchema } from "./validationSchema";
+import { addNewBook, useAppDispatch, useAppSelector } from "@/src/redux";
+import { bookProps } from "./AddBookForm.type";
+import { SelectField } from "@/src/components/Fields/SelectField";
+import { useBooksPageContext } from "@/src/pages/BooksPage/context";
+import { FilesService } from "@/src/services";
+import { apiUiMessage } from "@/src/types";
+import { AxiosError } from "axios";
+import { useErrorBoundary } from "@/src/hooks";
+
+const AddBookForm: FC = () => {
+  const { setIsAddBookModalOpen } = useBooksPageContext();
+  const { id } = useAppSelector((state) => state.userSlice.user);
+  const [bookImage, setbookImage] = useState("");
+  const [file, setFile] = useState<File | null>(null);
+  const [isLoading, setIsLoading] = useState(false);
+  const setErrorBoundary = useErrorBoundary();
+  const [error, setError] = useState<string | null>(null);
+  const dispatch = useAppDispatch();
+
+  const options = [
+    {
+      value: "1",
+      text: "Goin to read 📚",
+    },
+    {
+      value: "2",
+      text: "Reading 📖",
+    },
+    {
+      value: "3",
+      text: "Finished book 📗",
+    },
+  ];
+
+  const onHandleRemoveClick = () => {
+    setbookImage("");
+    setFile(null);
+  };
+
+  const onHandleSubmit = async (values: bookProps) => {
+    setIsAddBookModalOpen(false);
+    if (file) {
+      try {
+        setError(null);
+        setIsLoading(true);
+        const body: bookProps = {
+          book_statusId: values.book_statusId,
+          nameOfBook: values.nameOfBook,
+          author: values.author,
+          annotation: values.annotation,
+          imagePath: "",
+        };
+        if (file) {
+          const res = await new FilesService(
+            id,
+            file,
+            undefined,
+            setErrorBoundary
+          ).upload({
+            overwrite: false,
```

Comment:
Add the following value to the parameters of object => path: Path.books.
Path is an enum located at @/src/types. 
In this case, the url to the book image will be stored in cloudinary not in the posts folder, but in the books folder

Reply:
Fixed